packages/web-app-files/tests/unit/components/SideBar/Links/PrivateLinkItem.spec.js

```diff
@@ -0,0 +1,116 @@
+
+localVue.use(Vuex)
+localVue.use(DesignSystem)
+localVue.use(GetTextPlugin, {
+  translations: 'does-not-matter.json',
+  silent: true
+})
+
+const selectors = {
+  linkItem: '.oc-files-private-link-item',
+  copyToClipboardButton: '.oc-files-private-link-copy-url'
+}
+
+global.window = Object.create(window)
+const windowLocation = { href: 'http://localhost:9100' }
+Object.defineProperty(window, 'location', {
+  value: windowLocation
+})
+
+describe('PrivateLinkItem', () => {
+  function createStore({ isMounted = true } = {}) {
+    return new Vuex.Store({
+      getters: {
+        capabilities: function() {
+          return {
+            files: {
+              privateLinks: true
+            }
+          }
+        }
+      },
+      modules: {
+        Files: {
+          namespaced: true,
+          getters: {
+            highlightedFile: function() {
+              return {
+                isMounted: jest.fn(() => isMounted),
+                name: 'lorem.txt',
+                privateLink: 'some-link'
+              }
+            }
+          }
+        }
+      }
+    })
+  }
+
+  function getShallowWrapper(store) {
+    return shallowMount(PrivateLinkItem, {
+      localVue,
+      store,
+      stubs: {
+        'copy-to-clipboard-button': true
+      }
+    })
+  }
+
+  describe('when the private link functionality is enabled', () => {
+    it('should be visible', () => {
+      const store = createStore()
+      const wrapper = getShallowWrapper(store)
+
+      expect(wrapper.find(selectors.linkItem).exists()).toBeTruthy()
+    })
+
+    it('should render the copy-to-clipboard button', () => {
+      const store = createStore()
+      const wrapper = getShallowWrapper(store)
+
+      const copyToClipboardButtonElement = wrapper.find(selectors.copyToClipboardButton)
+
+      expect(copyToClipboardButtonElement.exists()).toBeTruthy()
+      expect(copyToClipboardButtonElement.attributes().label).toBe('Copy private link to clipboard')
+      expect(copyToClipboardButtonElement.attributes().successmsgtitle).toBe('Private link copied')
+      expect(copyToClipboardButtonElement.attributes().successmsgtext).toBe(
+        'The private link has been copied to your clipboard.'
+      )
+    })
+
+    describe('link item hyperlink', () => {
+      it('should render a hyperlink with href containing the scrollTo payload as a link if the highlighted file is mounted', () => {
+        const store = createStore()
+        const wrapper = getShallowWrapper(store)
+
+        expect(wrapper.find('a').attributes().href).toBe(
+          `${windowLocation.href}?scrollTo=${store.getters['Files/highlightedFile'].name}`
+        )
+      })
+
+      it('should render a hyperlink with href as a private link if the highlighted file is not mounted', () => {
+        const store = createStore({ isMounted: false })
+        const wrapper = getShallowWrapper(store)
+
+        expect(wrapper.find('a').attributes().href).toBe('some-link')
```

Comment:
is it possible to use a more specific selector than `a`, also add it to selectors.